boefjes/boefjes/plugins/kat_internetdb/normalize.py

```diff
@@ -0,0 +1,67 @@
+import json
+import logging
+from collections.abc import Iterable
+
+from boefjes.job_models import NormalizerMeta
+from octopoes.models import OOI, Reference
+from octopoes.models.ooi.dns.records import DNSPTRRecord
+from octopoes.models.ooi.dns.zone import Hostname
+from octopoes.models.ooi.findings import CVEFindingType, Finding
+from octopoes.models.ooi.network import IPPort, Network, PortState
+from octopoes.models.ooi.software import Software, SoftwareInstance
+
+DNS_PTR_STR = ".in-addr.arpa"
+
+
+def cpe_to_name_version(cpe: str) -> tuple:
```

Comment:
There is also a `get_name_from_cpe` in kat_binaryedge/services/normalize.py. I think it would be better to have a general cpe parsing function in plugins/helpers.py that is used by both this normalizer and the binaryedge normalizers. Functions like this should also have unit tests to make sure all edge cases are parsed correctly.